### DISCUSSION TITLE
[FEATURE REQUEST] Automatic minion installation for salt-ssh

### DISCUSSION THREAD

**Discussion by johnnybubonic** [saltstack/salt#67470] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
No, just a quality-of-life improvement thing.

**Describe the solution you'd like**
salt-ssh should have a flag to easily install the minion for the target(s). Ideally, if enabled it would:

1. Download and run the saltstack bootstrap script with relevant minion arguments
2. Generate a minion keypair
3. Accept that minion pubkey on the master (with prompting to the user perhaps?)

**Describe alternatives you've considered**
This can obviously be done via a state file and salt-ssh invoking that, but it seems a bit superfluous to need to do that when salt is invoking an SSH connection anyways.

**Additional context**
N/A

**Comment by network-shark** [comment]
I think this already possible with https://gitlab.com/saltstack/pop/heist ?

**Comment by johnnybubonic** [comment]
This seems to be restricted to `flat` roster types. I use a custom roster module.

**Comment by sagetherage** [comment]
If you are using salt-cloud saltify https://docs.saltproject.io/en/latest/topics/cloud/saltify.html could also be an option here, but unsure if you are using it or if it is helpful, simply giving more options.

**Comment by johnnybubonic** [comment]
@sagetherage I am not even using salt-cloud so saltify unfortunately isn't an option for me but I'm sure mentioning it will help someone searching the GH issues. 👍

**Comment by network-shark** [comment]
Didn't @Akm0d created a feature which let you install minions ?
